diff --git a/src/CivicrmMemberRoles.php b/src/CivicrmMemberRoles.php
index c6786c7..75c6147 100644
--- a/src/CivicrmMemberRoles.php
+++ b/src/CivicrmMemberRoles.php
@@ -67,7 +67,7 @@ class CivicrmMemberRoles {
     $this->civicrm = $civicrm;
     $this->config = $configFactory->get('civicrm_member_roles.settings');
     $this->entityTypeManager = $entityTypeManager;
-    $this->database = $database;
+    $this->database = $this->getDatabase();
   }

   /**
@@ -448,4 +448,15 @@ class CivicrmMemberRoles {
     return array_unique($roles);
   }

+  /**
+   * Gets the database.
+   *
+   * @return \Drupal\Core\Database\Connection
+   *   The database.
+   */
+  protected function getDatabase() {
+    $connection = \Drupal\Core\Database\Database::getConnection('default', 'civicrm');
+    return $connection;
+  }
+
 }
