diff --git a/js/google_analytics.debug.js b/js/google_analytics.debug.js
index 2df140a..9ba404f 100644
--- a/js/google_analytics.debug.js
+++ b/js/google_analytics.debug.js
@@ -13,6 +13,24 @@
 
   $(document).ready(function () {
 
+    if(Drupal.eu_cookie_compliance) {
+      //in case the eu_cookie_compliance events aren't working
+      const backupPref = setTimeout(postPreferencesLoadHandler, 2000);
+
+      function postPreferencesLoadHandler(response) {
+        clearTimeout(backupPref);
+
+        if(typeof sendPageView === 'function') {
+          sendPageView(Drupal);
+        }
+
+        ga("send", "pageview");
+      }
+
+      Drupal.eu_cookie_compliance('postPreferencesLoad', postPreferencesLoadHandler);
+      Drupal.eu_cookie_compliance('postStatusSave', postPreferencesLoadHandler);
+    }
+
     // Attach mousedown, keyup, touchstart events to document only and catch
     // clicks on all elements.
     $(document.body).on('mousedown keyup touchstart', function (event) {
diff --git a/js/google_analytics.js b/js/google_analytics.js
index 82b46cf..ebb883e 100644
--- a/js/google_analytics.js
+++ b/js/google_analytics.js
@@ -13,6 +13,24 @@
 
   $(document).ready(function () {
 
+    if(Drupal.eu_cookie_compliance) {
+      //in case the eu_cookie_compliance events aren't working
+      const backupPref = setTimeout(postPreferencesLoadHandler, 2000);
+
+      function postPreferencesLoadHandler(response) {
+        clearTimeout(backupPref);
+
+        if(typeof sendPageView === 'function') {
+          sendPageView(Drupal);
+        }
+
+        ga("send", "pageview");
+      }
+
+      Drupal.eu_cookie_compliance('postPreferencesLoad', postPreferencesLoadHandler);
+      Drupal.eu_cookie_compliance('postStatusSave', postPreferencesLoadHandler);
+    }
+
     // Attach mousedown, keyup, touchstart events to document only and catch
     // clicks on all elements.
     $(document.body).on('mousedown keyup touchstart', function (event) {
