diff --git a/image_widget_crop.install b/image_widget_crop.install
index 4026ecd..76f4e57 100644
--- a/image_widget_crop.install
+++ b/image_widget_crop.install
@@ -15,8 +15,8 @@ function image_widget_crop_requirements($phase) {
     $error = [];
     $config = \Drupal::config('image_widget_crop.settings');
     $files = [
-      'js' => $config->get('settings.library_url'),
-      'css' => $config->get('settings.css_url'),
+      'js' => $config->get('settings.library_url') ?? '',
+      'css' => $config->get('settings.css_url') ?? '',
     ];
 
     foreach ($files as $type => $file) {
