diff --git a/js/map.js b/js/map.js
index 1e929f0..03b23c7 100644
--- a/js/map.js
+++ b/js/map.js
@@ -102,11 +102,6 @@ function addEventListenersToFeature(feature, layer) {
   const properties = feature.properties;
 
   if(properties.is_enabled) {
-    if(properties.uri) {
-      layer.on('click', function() {
-        document.location.href = properties.uri;
-      });
-    }
 
     layer.bindTooltip(properties.tooltip || properties.name_long, {
       sticky: true,
