diff --git a/src/Plugin/field_group/FieldGroupFormatter/Details.php b/src/Plugin/field_group/FieldGroupFormatter/Details.php
index 3fa04af..34845de 100644
--- a/src/Plugin/field_group/FieldGroupFormatter/Details.php
+++ b/src/Plugin/field_group/FieldGroupFormatter/Details.php
@@ -51,6 +51,8 @@ class Details extends FieldGroupFormatterBase {
       $element['#attached']['library'][] = 'field_group/core';
     }
 
+    // Add details validation behaviour.
+    $element['#attached']['library'][] = 'field_group/details_validation';
   }
 
   /**
diff --git a/src/Plugin/field_group/FieldGroupFormatter/Tab.php b/src/Plugin/field_group/FieldGroupFormatter/Tab.php
index 4135a29..92c582d 100644
--- a/src/Plugin/field_group/FieldGroupFormatter/Tab.php
+++ b/src/Plugin/field_group/FieldGroupFormatter/Tab.php
@@ -66,6 +66,8 @@ class Tab extends FieldGroupFormatterBase {
 
     $element += $add;
 
+    // Add details validation behaviour.
+    $element['#attached']['library'][] = 'field_group/details_validation';
   }
 
   /**
diff --git a/src/Plugin/field_group/FieldGroupFormatter/Tabs.php b/src/Plugin/field_group/FieldGroupFormatter/Tabs.php
index 5218a43..ef101f0 100644
--- a/src/Plugin/field_group/FieldGroupFormatter/Tabs.php
+++ b/src/Plugin/field_group/FieldGroupFormatter/Tabs.php
@@ -59,6 +59,9 @@ class Tabs extends FieldGroupFormatterBase {
     if ($width_breakpoint = $this->getSetting('width_breakpoint')) {
       $element['#attached']['drupalSettings']['widthBreakpoint'] = $width_breakpoint;
     }
+
+    // Add details validation behaviour.
+    $element['#attached']['library'][] = 'field_group/details_validation';
   }
 
   /**
