diff --git a/core/modules/media/media.module b/core/modules/media/media.module
index 30b219e..bfbf61d 100644
--- a/core/modules/media/media.module
+++ b/core/modules/media/media.module
@@ -110,7 +110,7 @@ function media_theme_suggestions_media(array $variables) {
   $suggestions = [];
   /** @var \Drupal\media\MediaInterface $media */
   $media = $variables['elements']['#media'];
-  $sanitized_view_mode = strtr($variables['elements']['#view_mode'], '.', '_');
+  $sanitized_view_mode = strtr($variables['elements']['#view_mode'] ?? '', '.', '_');
 
   $suggestions[] = 'media__' . $sanitized_view_mode;
   $suggestions[] = 'media__' . $media->bundle();
